### PR TITLE
Workloadentry add should trigger push to the proxy

### LINKED ISSUE #28743: WorkloadEntry auto register race
We register the workload entry when it first connects to istiod, however if the workloadentry is not propagated back to the serviceentry store, then we will get no serviceinstances for this proxy. Then istiod can not geneerate right xds configs for this proxy.

And the workloadentry handler mainly triggers eds push, which will not update proxy at all.

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fix #28743

### CODE REVIEW

File: pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go [hunk 1]
```diff
@@ -519,6 +521,8 @@ func TestServiceDiscoveryWorkloadUpdate(t *testing.T) {
 	t.Run("add workload", func(t *testing.T) {
 		// Add a WLE, we expect this to update
 		createConfigs([]*config.Config{wle}, store, t)
+		//expectEvents(t, events, Event{kind: "xds", proxyIP: "2.2.2.2"})
```
Comment (shamsher31) [inline pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go:524]: You can remove this if it is not required anymore.

**Review by ramaraochavali — COMMENTED** [review]
LGTM

**Review by stevenctl — APPROVED** [review]

File: pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go (hunk 1 above)
Comment (stevenctl) [inline pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go:524]:
```suggestion

```

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go]: Thanks

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go]: THanks

**Comment by stevenctl** [comment]
@ramaraochavali needs @istio/wg-networking-maintainers-pilot stamp

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**